### PR TITLE
Add file builder API

### LINKED ISSUE #76: Write tests for context functions

### LINKED ISSUE #82: File builder API
This API is probably fine for now.  But if you grow more arguments, you might consider using the builder pattern, like this:
```rust
ctx.create(FileType::Regular)
    .name("loop1")
    .mode(0o644)
    .finish()?
```

_Originally posted by @asomers in https://github.com/musikid/pjdfstest/pull/79#discussion_r931734708_

### PR DESCRIPTION
**Pull request by musikid**
Add file builder API to conveniently create a file with a custom name/mode.
Also add tests for the context functions.

Close #82 
Close #76

### CODE REVIEW

File: rust/src/runner/context.rs [hunk 1]
```diff
@@ -286,3 +275,243 @@ impl Drop for TestContext {
...
+
+    #[test]
+    fn new_file() {
+        let current_umask = nix::sys::stat::umask(Mode::from_bits_truncate(ALLPERMS));
+        nix::sys::stat::umask(current_umask);
+
+        for ft in [
+            FileType::Regular,
+            FileType::Dir,
+            // FileType::Fifo,
```
Comment (asomers) [inline rust/src/runner/context.rs:494]: Did you mean to leave these comments here?

File: rust/src/runner/context.rs [hunk 2]
```diff
@@ -286,3 +275,243 @@ impl Drop for TestContext {
...
+                &path,
+            )
+            .map_err(|e| nix::Error::try_from(e).unwrap_or(nix::errno::Errno::UnknownErrno)),
+        }?;
+
+        Ok(path)
+    }
+
+    /// Create the file according to the provided information and open it.
+    pub fn open(self, oflags: OFlag) -> nix::Result<(PathBuf, RawFd)> {
```
Comment (asomers) [inline rust/src/runner/context.rs:343]: I think there are some tests that create a file with mode 0o444 and test that they can still write to it.  Those tests depend on the file being opened by the same syscall that created it.  They'll fail if you use this method, which closes the file and then reopens it.

Comment (musikid) [reply rust/src/runner/context.rs]: Thanks, I fixed it!

File: rust/src/runner/context.rs (hunk 1 above)
Comment (musikid) [reply rust/src/runner/context.rs:508]: Yes, I want to write tests for those file types, but they require specific handling.

**Pull request merged**